planner/optimizer/optimizer-core/src/main/java/eu/seaclouds/platform/planner/optimizer/util/YAMLoptimizerParser.java

```diff
@@ -483,9 +482,8 @@ private static Topology getApplicationTopologyRecursive(String elementName, Map<
       double hostPerformance = appInfoSuitableOptions.getCloudCharacteristics(elementName,
             YAMLmodulesOptimizerParser.getMeasuredPerformanceHost(elementName, groups)).getPerformance();
 
-      boolean elementCanScale= YAMLmodulesOptimizerParser.getScalabilityCapabilitiesOfModule(modules, elementName);
-            
-            
+      boolean elementCanScale = YAMLmodulesOptimizerParser.getScalabilityCapabilitiesOfModule(modules, elementName);
+
```

Comment:
Below, I think this [if-else](https://github.com/SeaCloudsEU/SeaCloudsPlatform/pull/195/files#diff-66a9293b4c1da2179fae3c61935da42cR498) could be refactor to something like.
```
if (!topology.contains(moduleReqName)) {
  topology = getApplicationTopologyRecursive(moduleReqName, (Map<String, Object>)
              modules.get(moduleReqName), topology, modules, groups, appInfoSuitableOptions);
}

double opProfileBetweenModules = YAMLgroupsOptimizerParser.getOpProfileWithModule(elementName,
                   moduleReqName, groups);	 
newelement.addElementCalled(
                   new TopologyElementCalled(topology.getModule(moduleReqName), opProfileBetweenModules));           
```